Album/ViewControllers/AuthenticationControllers/LoginViewController.m

```diff
@@ -19,64 +19,64 @@ @interface LoginViewController ()
+                                                                         if (error != nil) {
+                                                                         NSLog(@"User log in failed: %@", error.localizedDescription);
+                                                                         // Present auth error alert
+                                                                         [self authError];
+                                                                         } else {
+                                                                         NSLog(@"User logged in successfully");
+                                                                         // Segue to main home view
+                                                                         [self performSegueWithIdentifier:@"loginSegue" sender:nil];
+                                                                         }
+                                                                     }];
+} /* loginButton */
 
-- (void) authError {
-	UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Error" message:@"No such user exist!"
-	                            preferredStyle:(UIAlertControllerStyleAlert)];
-	// Create a cancel action
-	UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"Cancel"
-	                               style:UIAlertActionStyleCancel
-	                               handler:nil];
-	[alert addAction:cancelAction];
+- (void)authError {
+    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Error" message:@"No such user exist!"
+                                                            preferredStyle:(UIAlertControllerStyleAlert)];
+    // Create a cancel action
+    UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"Cancel"
+                                                           style:UIAlertActionStyleCancel
+                                                         handler:nil];
+    [alert addAction:cancelAction];
 
-	// Create an OK action
-	UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK"
-	                           style:UIAlertActionStyleDefault
-	                           handler:nil];
-	[alert addAction:okAction];
-	[self presentViewController:alert animated:YES completion:nil];
+    // Create an OK action
+    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK"
+                                                       style:UIAlertActionStyleDefault
+                                                     handler:nil];
+    [alert addAction:okAction];
+    [self presentViewController:alert animated:YES completion:nil];
 }
 - (IBAction)signUpButton:(id)sender {
-	[self performSegueWithIdentifier:@"signupSegue" sender:nil];
+    [self performSegueWithIdentifier:@"signupSegue" sender:nil];
 }
 
-- (IBAction) tap:(id) sender {
-	[self.view endEditing:YES];
+- (IBAction)tap:(id)sender {
+    [self.view endEditing:YES];
 }
```

Comment:
In general I'd create a `UIAlertAction` Helper method, probably in your Helpers class?

Reply:
This for the user to tap the screen and move on to the next input fields.